### DISCUSSION TITLE
New map engine

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#559] [Q&A] [answered]
**Cruiser 5** is being developed and will come with a _new map engine_:

- Map rotation in navigation

- https://github.com/devemux86/cruiser/discussions/558

- Clear rendering without artifacts or issues in overlapping areas

- OpenAndroMaps and Freizeitkarte maps / themes native support

- Motorider map theme (light / dark) by @MotoUKRider 

- Better battery performance

- No external dependencies

_Cruiser 5 is currently undergoing internal testing._

**Comment by devemux86** [comment]
@MotoUKRider

We will need to transfer the **Biker** and **Motorider** map themes to the new engine.
I will try the Biker map theme and let you know.

You probably just need to change some colors and parameters in Motorider themes.

**Reply by MotoUKRider** [reply]
@devemux86 Sure - if I can work on it with Cruiser Desktop or some other simple way before you're ready to unleash the APK, I'll try and get the colors/styling matched.  There aren't too many many differences between it and "Biker" now that we fixed a few rendering shortcomings (airports, train-tracks etc) so it *should* be a 1 :coffee: cup task :laughing:

**Reply by devemux86** [reply]
Cruiser 5 desktop will also use the same map engine.

**Reply by DIrgi** [reply]
I would love to see there a point "own map design"
So far I have to add/load  it via map/external map design. It could be setup once in options (the path) and then always be available for switching and activating under the point "own map design"

**Reply by devemux86** [reply]
> So far I have to add/load it via map/external map design. It could be setup once in options (the path) and then always be available for switching and activating under the point "own map design"

@DIrgi Thanks for the suggestion.

Can you please also write it in a new topic so it's not forgotten here?

**Comment by devemux86** [comment]
It's also probably time to set the minimum Android requirement to Android 6.

**Reply by DIrgi** [reply]
Well, 6 started in 10/2015 ...long time ago so....yes...fair enough at least....

**Reply by 0709wiwiwi** [reply]
That it will come to that is clear. Since I'm not much of a throwaway of material that works, I still use my A5.1 ( was A 4.4) phone on the bike. Great bright screen and LOUD speaker. The end will come one day as there is no replacement battery available anymore. -;) Recuperation of ? Through combustion and toxic smoke, some precious metals are then recovered for alms. Due to no longer compatible Android changes the large company software developers or men with ties encourage such pollution.

**Comment by DIrgi** [comment]
I would love to see the polygon shade possibility to mark and display limited areas i.e. green zoneson the map
Remember we had a discussion 2 month ago where you said that the function is available but not in use so far.

**Reply by devemux86** [reply]
@DIrgi if you mean loading GeoJSON files, it will be available:

- https://github.com/devemux86/cruiser/discussions/506

**Comment by DIrgi** [comment]
Will Cruiser 5 be "more 3D"?
On 2D view with hill shading, Cruiser has already somehow a "3D" effect.
Switching to 3D view (btw. can/will it be possible to switch the angle - manually?), 
hill shading loosing it's strength or better I assume an elevation model is missing.

See screenshot from Garmin.
![image](https://github.com/devemux86/cruiser/assets/57968526/d0733686-5248-4e09-8697-337bbcc1c366)

Do you have any statement regarding elevation models and "read 3D effect"?

**Reply by devemux86** [reply]
There will be no 3D effect in the map rotation, at least at the beginning.

Hillshading will use digital elevation model data in [HGT](http://wiki.openstreetmap.org/wiki/SRTM#Format) format.
They can be downloaded like the map files and be used **offline**.

I updated the hillshading topic with some useful links:
- https://github.com/devemux86/cruiser/discussions/558

**Comment by abdullahO2** [comment]
Thank you for the great work on Cruiser 5 and the exciting new features. I have a few suggestions that I believe would greatly enhance the user experience:

- Support for adding custom raster online maps (TMS, WMS) by URL.
- Underlay/Overlay layer support.
- Easier off-road navigation with straight-line routing:
 https://github.com/devemux86/cruiser/discussions/204


Looking forward to the new version. Thanks again for your hard work!

**Comment by devemux86** [comment]
@MotoUKRider

Can you send me an email to discuss how we could proceed with the themes?
You can use the email on [Cruiser's Google Play page](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

**Comment by kds112** [comment]
Any timetable plans for a first beta release...? 😉

**Reply by devemux86** [reply]
It will be much sooner than expected. :slightly_smiling_face: 

We could test and optimize the new map engine in the next 4.X Beta releases (which will also keep OpenGL).
Cruiser 5 will be released with the new map engine only and its new features can then be included gradually.

**Comment by devemux86** [comment]
Screenshots of **Motorider** theme for Cruiser 5 prepared by @MotoUKRider can be seen here:

- https://github.com/mapsforge/mapsforge/issues/1483#issuecomment-2169386524

**Comment by MotoUKRider** [comment]
An extra treat for this group: This is a comparison of the current (OpenGL/VTM) theme vs the forthcoming Mapsforge-based theme:

![cruiser-vtm-vs-mapsforge](https://github.com/devemux86/cruiser/assets/125696109/0e4a72a3-2ad8-4b04-b553-ebdc87e2eb74)

For info, my current light/dark themes in Cruiser v4 are based on a heavily-modified version of "Biker", and these new light+dark are a significant re-write of the "Default" Mapsforge XML. I have [previously described the tools+process used here](https://github.com/devemux86/cruiser/discussions/176#discussioncomment-5998471).

**Reply by devemux86** [reply]
@MotoUKRider

Excellent work as always! :+1:

**Comment by DolomitesTom** [comment]
Will the Cruiser 5 development be a Greenfield approach or keep existing functions?

**Reply by devemux86** [reply]
@DolomitesTom 

Cruiser 5 continues with all existing functions, just different map engine.
It will also be easier to implement new features. :slightly_smiling_face:

**Comment by MotoUKRider** [comment]
One of the aspects of re-writing a theme for native Mapsforge is that of "detail" - typically, the more you zoom in, the more things appear like minor roads, dirt-tracks, buildings and POIs etc. But what if you want to follow a Primary/Secondary road in a remote area at a lower zoom level?

Layers - that's the answer!  In the same way that you can presently turn on/off "Fuel", "Food" and other POIs, we've been working on a test layer called `Extra road detail` which allows the road "ref" labels to appear earlier.  Here's a good example from the south of the UK where a more remote area may benefit from more road detail:

![cruiser-mapsforge-detail-2](https://github.com/devemux86/cruiser/assets/125696109/43d24b97-62aa-4398-a8ab-089ed7f3bd53)

...and here's the opposite scenario - London (and any big city) completely swamped with road labels:

![cruiser-mapsforge-detail-1](https://github.com/devemux86/cruiser/assets/125696109/e38b8172-3c19-4258-b7fa-428f552f2f81)

The extra detail brings "Primary" roads in at level 10 (vs 12) and "Secondary" at 11 (vs 12).  The "Dark" theme will offer the same function.

Other map features (eg: special land areas/captions not important for daily navigation) may be controlled in this way and the final themes will be up here to "grab" and edit yourself - I'm always happy to answer questions about styling either here or on the [Mapsforge project](https://github.com/mapsforge/mapsforge), just tag me if you need me to reply.

**Reply by DIrgi** [reply]
@MotoUKRider Hi, don't get me wrong, I know it's hard and exzellent work but in your new version I feel your road naming very "overwhelming" or "too bold" although it's more modern to the old the old one was not so intrusive like the new one.

**Comment by Zaptrouiyerxi** [comment]
Hello,

Very naive question, pls be kind and redirect me if this is the wrong spot:
Is/will there be any way to dumb down the look of maps, way down to something like Google Maps navigation mode?

Im working on a motorcycle dash (using a Valve Steam Deck) and I only have about 1000px height with lots of things on the rest of the display (tach, throttle, fuel, gear, rear cam, etc).
![image](https://github.com/user-attachments/assets/a56abf14-9607-448d-8d73-52c8d6eec96f)
Not necessarily this extremely stripped down either.

Is this a realistic thing now or in the near term, or DIY?  Or not?
Have I missed some API to get simple "next instructions" from?  E.G. "Right turn at intersection in 500m"

**Reply by devemux86** [reply]
Hi @Zaptrouiyerxi,

Do you mean Cruiser on Android watch?

Or Cruiser on the phone to display the map with less details?

**Reply by Zaptrouiyerxi** [reply]
Cruiser on Linux.  My plan is to build my app(s) on a Valve Steam Deck which runs some Arch Linux flavor (but can run other distros too).  I'd integrate Cruiser on demand (some handlebar button and/or phone API call) with the rest of the dash.

A lot like some modern automotive dashes have multiple dash modes - normal / racetrack / navigation / ...
![image](https://github.com/user-attachments/assets/739e15bd-f215-415e-bd64-4dd8a7accee7)

a rough example.

![image](https://github.com/user-attachments/assets/2c650f8c-5f89-4a21-aa2f-4cff368401cf)
[steam deck specs ](https://www.steamdeck.com/en/tech)   
Ripping out the extraneous gaming hardware, obviously.  The specs are good enough to make it run everything (incl analog telemetry).  The alternative is to run everything on something like a raspberry pi, and display everything on a dedicated display part - phone, tablet.

**Reply by devemux86** [reply]
@Zaptrouiyerxi I work on Linux!

Desktop Cruiser is a Java application, so it runs everywhere.
Arch Linux users can also install Cruiser from the AUR: https://aur.archlinux.org/packages/cruiser

Installation instructions are here:
https://wiki.openstreetmap.org/wiki/Cruiser#Installation

**Reply by Zaptrouiyerxi** [reply]
Yes I have everything bookmarked for soon :)

So no way to change Cruiser's underlying maps' styling?

**Reply by devemux86** [reply]
@Zaptrouiyerxi Online maps have fixed styling.

Offline maps have customized styling via the selected render theme.

Cruiser has some included render themes for the Mapsforge offline maps,
like the new Motorider theme seen above and here:
- https://github.com/mapsforge/mapsforge/issues/1483#issuecomment-2169386524

[OpenAndroMaps](https://www.openandromaps.org/en) and [Freizeitkarte](https://www.freizeitkarte-osm.de/android/en/index.html) offline maps also provide their own render themes.

You can also change or create a render theme ([instructions](https://github.com/mapsforge/mapsforge/blob/master/docs/Rendertheme.md)).

**Reply by Zaptrouiyerxi** [reply]
Thank you devemux :)

**Comment by devemux86** [comment]
Cruiser 5 with the new map engine will use a new intelligent map display.

The more you use Cruiser, the faster it becomes with instant map display without delay.

Better user experience with the goal of zero battery consumption from the map. :slightly_smiling_face:

**Reply by DolomitesTom** [reply]
Update for testing comes when?

**Reply by devemux86** [reply]
> Update for testing comes when?

Very soon. :slightly_smiling_face:

**Comment by devemux86** [comment]
Cruiser 5 new map engine and the new Motorider map theme (light / dark) by @MotoUKRider are available in:

- https://github.com/devemux86/cruiser/discussions/610

**Comment by DolomitesTom** [comment]
can you please give a short overview why there is a Cruiser 4 and 5 lane? what are the main differences/benefits and will they exist parallel all the time?

**Reply by devemux86** [reply]
@DolomitesTom 

- Cruiser 4 was named mainly because of the [offline curvy routing](https://github.com/devemux86/cruiser/discussions/254)

- Cruiser 5 is named mainly because of the new map engine

> will they exist parallel all the time?

Cruiser 5 replaces Cruiser 4 after installation and advances the favorites, routes, tracks, profiles to [version 2](https://github.com/devemux86/cruiser/discussions/585).

If you want to go back, you will need to import the backup of my collection again.

**Comment by DolomitesTom** [comment]
And are maps for Cruiser 5 not rotatable or is this a setting?

**Reply by devemux86** [reply]
@DolomitesTom 

Map rotation is enabled automatically in:
- Navigation
- Follow location

Route planning does not need map rotation, it uses the default "North is up" orientation like paper maps.

**Reply by DolomitesTom** [reply]
I found that very important, also when stopping or or pausing navigation beeing able to rotate the map. Can this be activated? I found that also a big disadvantage in Kurviger, one main reason why I didn't proceed with it.

**Reply by DolomitesTom** [reply]
> Route planning does not need map rotation

I disagree

**Reply by devemux86** [reply]
> I found that very important, also when stopping or or pausing navigation beeing able to rotate the map. Can this be activated?

Map rotation **gesture** is currently not available, it may come in the future.

**Reply by DolomitesTom** [reply]
> Map rotation gesture is currently not available, it may come in the future.

ok thanks, any guess when? And can you please place a link with the latest 4.1.x? I would like to wait with further testing until it is available. Map rotation is really important for me.

**Reply by devemux86** [reply]
@DolomitesTom 

If you don't like to advance to Cruiser 5, you can stay with Cruiser 4.0.2 (available on [Releases](https://github.com/devemux86/cruiser/releases)).

(going from 5 to 4 you will need to import the backup of my collection again)

**Reply by DolomitesTom** [reply]
thanks, have made a backup and tried Cruiser 5 only on my old device. I hope the mentioned topics will come soon, then I try it again.

**Comment by devemux86** [comment, marked as answer]
- Available in https://github.com/devemux86/cruiser/discussions/641
